### ISSUE TITLE
`(vtd/fragment (vtd/navigator "<Foo/>"))` crashes

### ISSUE THREAD

**Issue by borkdude** [mudge/riveted#4]
The `fragment` function crashes when given a navigator to a self-closing tag:

```
(vtd/fragment (vtd/navigator "<Foo/>"))
;;=> java.lang.NegativeArraySizeException
```

**Comment by mudge** [comment]
Hi @borkdude,

This seems to be a [bug in `fragment`](https://github.com/mudge/riveted/blob/8348d9ae916aea8e79ca4ec434d02acce2e3ea1d/src/riveted/core.clj#L184) where I'm not checking for empty elements: the underlying [`getContentFragment`](https://vtd-xml.sourceforge.io/javadoc/com/ximpleware/VTDNav.html#getContentFragment--) method in VTD will return `-1` if the element is empty.

This should be simple to fix but what would you expect the result to be: an empty string or `nil`?

**Comment by borkdude** [comment]
For consistency, the same result as 

```
(vtd/fragment (vtd/navigator "<Foo></Foo>"))
```

which is now an empty string, but nil would also make sense to me.

**Comment by mudge** [comment]
This should now be fixed in 0.1.1, can you please try it out and let me know if that fixes your issue?

**Comment by borkdude** [comment]
Actually we use a similar function, but with `toRawString` in which I ran into this error. I updated it now too:

```
(when-let [nav (.nav navigator)]
    (let [r (.getContentFragment nav)]
      (if (= r -1)
        ""
        (.toRawString nav
                      ^long (long r)
                      ^long (bit-shift-right r 32)))))
```

and both work correctly now. Thanks.
